biz.aQute.bndlib.tests/test/test/AnalyzerTest.java

```diff
@@ -263,6 +265,26 @@ public void testBundleClasspathWithVersionedExports() throws Exception {
 
 	}
 
+	public void testNegatedPackagesFiltering() throws Exception {
+		Builder b = new Builder();
+		try {
+			b.setIncludePackage("test.packageinfo.*");
+			b.setProperty("X-Not-Versioned", "${packages;ANNOTATED;!org.osgi.annotation.versioning.Version}");
+			b.addClasspath(new File("bin_test"));
+			Jar jar = b.build();
+			Manifest m = jar.getManifest();
+			m.write(System.err);
+			assertTrue(b.check());
+			Parameters notVersioned = Domain.domain(m)
+				.getParameters("X-Not-Versioned");
+			assertThat(notVersioned).containsKeys("test.packageinfo", "test.packageinfo.both_no_version",
+				"test.packageinfo.nopackageinfo", "test.packageinfo.ref");
```

Comment:
Should there be some keys that are tested to not be contained? `doesNotContain`?

Reply:
done